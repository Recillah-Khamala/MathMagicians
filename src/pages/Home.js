import React from 'react';
import styles from './Home.module.css';

const Home = () => (
  <div className={styles.home}>
    <h1>Welcome to our page!</h1>
    <p>
      Lorem, ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia,
      molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum
      numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium
      optio,
    </p>
    <p>
      Lorem, ipsum dolor sit amet consectetur adipisicing elit. Inventore
      veritatis animi excepturi explicabo reiciendis, quaerat vitae saepe quam
      impedit expedita suscipit ut, blanditiis neque voluptate voluptatibus.
      Veritatis excepturi dolor ut!
    </p>
  </div>
);

export default Home;
