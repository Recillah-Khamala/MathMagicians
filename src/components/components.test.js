import { render, screen } from "@testing-library/react";
import Quote from "../pages/Quotes";
import Home from "../pages/Home";
import Calculator from './calculator';

describe("testing component", () => {
  test("testing Homepage", () => {
    const home = render(<Home />);
    expect(home).toMatchSnapshot();
  });
  test("testing quote", () => {
    const quote = render(<Quote />);
    expect(quote).toMatchSnapshot();
  });
  test('testing calculator page', async() => {
    render(<Calculator />);
    const btn = await screen.findAllByRole('button');
    expect(btn).toHaveLength(19);
  });
});
