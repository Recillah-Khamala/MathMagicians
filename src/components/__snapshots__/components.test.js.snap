// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing component testing Homepage 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home"
      >
        <h1>
          Welcome to our page!
        </h1>
        <p>
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio,
        </p>
        <p>
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Inventore veritatis animi excepturi explicabo reiciendis, quaerat vitae saepe quam impedit expedita suscipit ut, blanditiis neque voluptate voluptatibus. Veritatis excepturi dolor ut!
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home"
    >
      <h1>
        Welcome to our page!
      </h1>
      <p>
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio,
      </p>
      <p>
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Inventore veritatis animi excepturi explicabo reiciendis, quaerat vitae saepe quam impedit expedita suscipit ut, blanditiis neque voluptate voluptatibus. Veritatis excepturi dolor ut!
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing component testing quote 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="quotes"
      >
        <p>
          Mathematics is not about numbers, equations, computations, or algorithms: it is about understaning. -William Paul Thurston
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="quotes"
    >
      <p>
        Mathematics is not about numbers, equations, computations, or algorithms: it is about understaning. -William Paul Thurston
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
